R/extract_coefficients.R

```diff
@@ -0,0 +1,127 @@
+
+# Extract exposure coefficients for specific exposure and outcomes, i.e. metabolites (?)#
+getExp.coef <- function(object, outcome, exposure) {
+
+  # object: output of net.coupler
+  # outcome: list of metabolites to evaluate, e.g. all
+  # exposure: name of considered exposure variable
+
+  cat("*************************************************************************************************** \n")
+  cat("This function produces a table of effect estimates of the exposure (betas, SE, t-value, p-value) on \n")
+  cat("a selected (set of) outcome(s) [specified as character-vector of variable-names] for all possible   \n")
+  cat("causal models based on conditional independence criteria encoded in the input-network => MULTISET OF\n")
+  cat("POSSIBLE EFFECTS PER EXPOSURE-OUTCOME PAIRS: NUMBER OF POSSIBLE EFFECTS ~ SIZE OF THE ADJECENCY SET!\n")
+  cat("***************************************************************************************************")
+
+  # define empty dataframes for output:
+
+  mm_coef <- structure(list(
+    Model = as.character(),
+    Nbmds = as.numeric(),
+    Outcome = as.character(),
+    Exposure = as.character(),
+    Covariables = as.character(),
+    Estimate = as.numeric(),
+    SE = as.numeric(),
+    tval = as.numeric(),
+    P = as.numeric()
+  ),
+  class = "data.frame"
+  )
+
+  mm_coef_temp <- structure(list(
+    Model = as.character(),
+    Nbmds = as.numeric(),
+    Outcome = as.character(),
+    Exposure = as.character(),
+    Covariables = as.character(),
+    Estimate = as.numeric(),
+    SE = as.numeric(),
+    tval = as.numeric(),
+    P = as.numeric()
+  ),
+  class = "data.frame"
+  )
+
+  # for each exposure-outcome pair: get exposure coefficients from all possible models and write to table:
+
+  for (j in outcome) {
+    mm_coef_temp <- data.frame(getExp.coef.peroutcome(object = object, outcome = j, exposure = exposure))
+    mm_coef <- bind_rows(mm_coef, mm_coef_temp)
+  }
+
+  return(mm_coef)
+}
+
+# # B. Get exposure coefficients for a a group of outcomes: e.g. all network-variables
```

Comment:
This function is different from the one above. Check the commit messages for more detail.